synapse/storage/_base.py

```diff
@@ -650,20 +653,29 @@ def _simple_select_list_txn(cls, txn, table, keyvalues, retcols):
                 column names and values to select the rows with, or None to not
                 apply a WHERE clause.
             retcols (iterable[str]): the names of the columns to return
+            order (str): order the select by this column
+            order_asc (bool): order ascending (if order specified). Default True.
         """
+        query_values = []
         if keyvalues:
             sql = "SELECT %s FROM %s WHERE %s" % (
                 ", ".join(retcols),
                 table,
                 " AND ".join("%s = ?" % (k, ) for k in keyvalues)
             )
-            txn.execute(sql, keyvalues.values())
+            query_values.extend(keyvalues.values())
```

Comment:
For performance reasons, I'd probably rely on variable hoisting and do:

```python
if keyvalues:
    ...
    query_values = keyvalues.values()
else:
    query_values = []
```

Reply:
wfm!